docs/source/conf.py

```diff
@@ -16,8 +16,16 @@
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
 #
+from __future__ import annotations
+
+import collections.abc
 import os
 import sys
+from typing import TYPE_CHECKING, cast
+
+if TYPE_CHECKING:
+    from sphinx.application import Sphinx
+    from sphinx.util.typing import Inventory
```

Comment:
Not sure if the guard is necessary. If this is running, Sphinx is installed?

Reply:
Ruff does this automatically for any objects that are only found in type hints for some reason

Reply:
Ahh right.

Reply:
Hah, I did it manually out of habit. I like denoting typing-only imports